apps/marketplace/actions/teamActions.js

```diff
@@ -0,0 +1,158 @@
+  BUYER_TEAM_MEMBERS_SUCCESS,
+  CREATE_TEAM_SUCCESS,
+  GET_TEAM_SUCCESS,
+  SAVE_TEAM_SUCCESS,
+  TEAMS_OVERVIEW_SUCCESS,
+  USER_ORGANISATION
+} from '../constants/constants'
+import { GENERAL_ERROR } from '../constants/messageConstants'
+import dmapi from '../services/apiClient'
+import { clearErrorMessages, sendingRequest, setErrorMessage } from './appActions'
+
+export const handleCreateTeamSuccess = response => ({
+  type: CREATE_TEAM_SUCCESS,
+  team: response.data
+})
+
+export const createTeam = () => (dispatch, getState) => {
+  dispatch(sendingRequest(true))
+  return dmapi({
+    url: '/team/create',
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'X-CSRFToken': getState().app.csrfToken
+    }
+  }).then(response => {
+    if (!response || response.error) {
+      const errorMessage = response.data.message ? response.data.message : GENERAL_ERROR
+      dispatch(setErrorMessage(errorMessage))
+    } else {
+      dispatch(handleCreateTeamSuccess(response))
+    }
+    dispatch(sendingRequest(false))
+    return response
+  })
+}
+
+export const handleGetTeamSuccess = response => ({
+  type: GET_TEAM_SUCCESS,
+  team: response.data
+})
+
+export const getTeam = teamId => (dispatch, getState) => {
+  dispatch(sendingRequest(true))
+  return dmapi({
+    url: `/team/${teamId}`,
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      'X-CSRFToken': getState().app.csrfToken
+    }
+  }).then(response => {
+    if (!response || response.error) {
+      const errorMessage = response.data.message ? response.data.message : GENERAL_ERROR
+      dispatch(setErrorMessage(errorMessage))
+    } else {
+      dispatch(handleGetTeamSuccess(response))
+    }
+    dispatch(sendingRequest(false))
+    return response
+  })
+}
+
+export const handleSaveTeamSuccess = response => ({
+  type: SAVE_TEAM_SUCCESS,
+  team: response.data.team
+})
+
+export const saveTeam = team => (dispatch, getState) => {
+  dispatch(sendingRequest(true))
+  dispatch(setErrorMessage(null))
+  return dmapi({
+    url: `/team/${team.id}/update`,
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'X-CSRFToken': getState().app.csrfToken
+    },
+    data: JSON.stringify(team)
+  }).then(response => {
+    if (!response || response.error) {
+      const errorMessage = response.data.message ? response.data.message : GENERAL_ERROR
+      dispatch(setErrorMessage(errorMessage))
+    } else {
+      dispatch(handleSaveTeamSuccess(response))
+    }
+    dispatch(sendingRequest(false))
+    return response
+  })
+}
+
+export const findTeamMember = (keywords, exclude) => (dispatch, getState) => {
+  const params = { keywords, exclude: exclude ? exclude.join(',') : '' }
+
+  dispatch(sendingRequest(true))
+  return dmapi({
+    url: '/team/members/search',
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      'X-CSRFToken': getState().app.csrfToken
```

Comment:
No harm, but you shouldn't need this for a GET.

Reply:
removed